src/Compilers/CSharp/Test/Emit/Emit/LocalStateTracing/LocalStateTracingTests.cs

```diff
@@ -0,0 +1,5898 @@
+        {
+            int a = 1;    
+        }
+
+        F();
+
+        Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload();
+    }
+}
+") + CSharpInstrumentationChecker.InstrumentationHelperSource;
+
+            var verifier = CompileAndVerify(
+                source,
+                options: TestOptions.UnsafeDebugExe,
+                emitOptions: EmitOptions.Default
+                    .WithInstrumentationKinds(ImmutableArray.Create(InstrumentationKind.LocalStateTracing, InstrumentationKind.TestCoverage)),
+                verify: s_verification,
+                targetFramework: s_targetFramework);
+
+            verifier.VerifyMethodBody("C.<Main>g__F|0_0", @"
+{
+  // Code size       41 (0x29)
+  .maxstack  3
+  .locals init (Microsoft.CodeAnalysis.Runtime.LocalStoreTracker V_0,
+                int V_1) //a
+  // sequence point: <hidden>
+  IL_0000:  ldtoken    ""void C.Main()""
+  IL_0005:  ldtoken    ""void C.<Main>g__F|0_0()""
+  IL_000a:  call       ""Microsoft.CodeAnalysis.Runtime.LocalStoreTracker Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogLambdaEntry(int, int)""
+  IL_000f:  stloc.0
+  .try
+  {
+    // sequence point: {
+    IL_0010:  nop
+    // sequence point: int a = 1;
+    IL_0011:  ldloca.s   V_0
+    IL_0013:  ldc.i4.1
+    IL_0014:  dup
+    IL_0015:  stloc.1
+    IL_0016:  ldc.i4.1
+    IL_0017:  call       ""void Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogLocalStore(uint, int)""
+    IL_001c:  nop
+    // sequence point: }
+    IL_001d:  leave.s    IL_0028
+  }
+  finally
+  {
+    // sequence point: <hidden>
+    IL_001f:  ldloca.s   V_0
+    IL_0021:  call       ""void Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogReturn()""
+    IL_0026:  nop
+    IL_0027:  endfinally
+  }
+  // sequence point: }
+  IL_0028:  ret
+}
+");
+        }
+
+        [Fact]
+        public void EmitDifference()
+        {
+            var source = WithHelpers(@"
+class C
+{
+    static void F()
+    {
+        int a = 0xFFFF;
+
+        static void LF() { int b = 0xEEEE; }
+
+        LF();
+    }
+}
+");
+
+            var compilation0 = CreateCompilation(source, options: TestOptions.UnsafeDebugDll, targetFramework: s_targetFramework);
+            var compilation1 = compilation0.WithSource(source);
+
+            var f0 = compilation0.GetMember("C.F").GetPublicSymbol();
+            var f1 = compilation1.GetMember("C.F").GetPublicSymbol();
+
+            using var md0 = ModuleMetadata.CreateFromImage(compilation0.EmitToArray());
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, debugInformationProvider: _ => default);
+
+            var diff = compilation1.EmitDifference(
+                generation0,
+                edits: ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, f0, f1, instrumentation: new MethodInstrumentation() { Kinds = ImmutableArray.Create(InstrumentationKind.LocalStateTracing) })));
+
+            diff.VerifyIL(@"
```

Comment:
I don't see the instrumentation in the IL below, should it be there?

Reply:
All the calls `call       0x060000XY` are to the instrumentation helpers. Also the `endfinally` indicates the code has been instrumented. Unfortunately we don't currently have ILVerify helpers that would resolve the tokens for ENC delta IL.